rpc/src/responses/chains/blocks/invalid_blocks_in_chain.rs

```diff
@@ -6,148 +6,188 @@ use serde_json::{json, Value};
-				"errors":{}
+            mock_kind,
+            mock_id,
+            mock_invalid_block_hash,
+            mock_error,
+            mock_extra_key,
+            mock_extra_value
+        );
+
+        let trimmed_string = trim_and_remove_whitespace_from_string(mock_error_response);
+        trimmed_string
+    }
+
+    fn format_response_data_as_string(
+        mock_block: &str,
+        mock_level: i32,
+        mock_error_response: &str,
+    ) -> String {
+        let mock_response = format!(
+            r#"[{{
+                "block":"{}",
+                "level":{},
+                "errors":{}
 			}}]"#,
-			mock_block, mock_level, mock_error_response
-		);
-
-		mock_response.replace('\n', "").replace('\t', "")
-	}
-
-	#[test]
-	fn get_invalid_blocks_in_chain_from_response_empty_fails() {
-		let mock_response = "";
-
-		let invalid_blocks_response = InvalidBlocksInChainResponse::from_response_str(mock_response);
-		assert!(invalid_blocks_response.is_err());
-	}
-
-	#[test]
-	fn get_invalid_blocks_in_chain_from_response_ok() {
-		let mock_block = "blockId1";
-		let mock_level = 1;
-
-		let mock_kind = "permanent";
-		let mock_id = "validator.invalid_block";
-		let mock_invalid_block_hash = "blockId1";
-		let mock_error = "cannot_parse_operation";
-		let mock_extra_key = "operation";
-		let mock_extra_value = "operationHash1";
-
-		let mock_error_response = get_mock_error_response(
-			mock_kind,
-			mock_id,
-			mock_invalid_block_hash,
-			mock_error,
-			mock_extra_key,
-			mock_extra_value,
-		);
-
-		let mock_response = get_mock_response(mock_block, mock_level, &mock_error_response);
-
-		let invalid_blocks_response = InvalidBlocksInChainResponse::from_response_str(&mock_response);
-		assert!(invalid_blocks_response.is_ok());
-
-		let mut invalid_blocks = invalid_blocks_response.unwrap().invalid_blocks.into_vec();
-		assert!(invalid_blocks.len() == 1);
-
-		let invalid_block = invalid_blocks.pop().unwrap();
-		let invalid_block_array_str = format!("[{}]", invalid_block);
-		assert_eq!(invalid_block_array_str, mock_response);
-	}
-
-	#[test]
-	fn get_invalid_blocks_in_chain_from_malformed_response_fails() {
-		let mock_block = "blockId2";
-		let mock_level = 2;
-
-		// Mock response without errors field, which is required for all type of errors
-		let mock_response = format!(
-			r#"[{{
-				"block":"{}",
-				"level":{},
-			}}]"#,
-			mock_block, mock_level
-		);
-
-		let invalid_blocks_response = InvalidBlocksInChainResponse::from_response_str(&mock_response);
-		assert!(invalid_blocks_response.is_err());
-	}
+            mock_block, mock_level, mock_error_response
+        );
+
+        let trimmed_string = trim_and_remove_whitespace_from_string(mock_response);
+        trimmed_string
+    }
+
+    fn trim_and_remove_whitespace_from_string(pre_format_string: String) -> String {
+        pre_format_string
+            .replacen(' ', "", usize::MAX)
+            .replace('\n', "")
+            .replace('\t', "")
+    }
```

Comment:
The formatting would mess up randomly due to non-tab/newline whitespace, so although seemingly a little redundant.